src/Centreon/Infrastructure/MetaServiceConfiguration/Repository/MetaServiceConfigurationRepositoryRDB.php

```diff
@@ -0,0 +1,333 @@
+                    ON ag.acl_group_id = agcr.acl_group_id
+                LEFT JOIN `:db`.acl_group_contactgroups_relations agcgr
+                    ON ag.acl_group_id = agcgr.acl_group_id
+                LEFT JOIN `:db`.contactgroup_contact_relation cgcr
+                    ON cgcr.contactgroup_cg_id = agcgr.cg_cg_id
+                    AND cgcr.contact_contact_id = :contact_id"
+            );
+        }
+
+        // Search
+        $searchRequest = $this->sqlRequestTranslator->translateSearchParameterToSql();
+        $request .= !is_null($searchRequest) ? $searchRequest : '';
+
+        // Sort
+        $sortRequest = $this->sqlRequestTranslator->translateSortParameterToSql();
+        $request .= !is_null($sortRequest)
+            ? $sortRequest
+            : ' ORDER BY ms.meta_name ASC';
+
+        // Pagination
+        $request .= $this->sqlRequestTranslator->translatePaginationToSql();
+        $statement = $this->db->prepare($request);
+
+        foreach ($this->sqlRequestTranslator->getSearchValues() as $key => $data) {
+            $type = key($data);
+            $value = $data[$type];
+            $statement->bindValue($key, $value, $type);
+        }
+        if ($contactId !== null) {
+            $statement->bindValue(':contact_id', $contactId, \PDO::PARAM_INT);
+        }
+        $statement->execute();
+
+        $result = $this->db->query('SELECT FOUND_ROWS()');
+        if ($result !== false && ($total = $result->fetchColumn()) !== false) {
+            $this->sqlRequestTranslator->getRequestParameters()->setTotal((int) $total);
+        }
+
+        $metaServicesConfigurations = [];
+        while (($record = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            $metaServicesConfigurations[] = MetaServiceConfigurationFactoryRdb::create($record);
+        }
+        return $metaServicesConfigurations;
+    }
+
+    /**
+     * @inheritDoc
+     * @throws AssertionFailedException
+     */
+    public function findById(int $metaId): ?MetaServiceConfiguration
+    {
+        return $this->findByIdRequest($metaId, null);
+    }
+
+    /**
+     * @inheritDoc
+     * @throws AssertionFailedException
+     */
+    public function findByIdAndContact(int $metaId, ContactInterface $contact): ?MetaServiceConfiguration
+    {
+        return $this->findByIdRequest($metaId, $contact->getId());
+    }
+
+    /**
+     * Find a category by id and contact id.
+     *
+     * @param int $metaId Id of the meta service configuration to be found
+     * @param int|null $contactId Contact id related to host categories
+     * @return MetaServiceConfiguration|null
+     * @throws AssertionFailedException
+     */
+    private function findByIdRequest(int $metaId, ?int $contactId): ?MetaServiceConfiguration
+    {
+        if ($contactId === null) {
+            $statement = $this->db->prepare(
+                $this->translateDbName(
+                    "SELECT ms.meta_id,
```

Comment:
`SELECT` could be construct before `if` statement
Then, if contact id is set, add `JOIN`